### DISCUSSION TITLE
Can't add primary IP for a device

### DISCUSSION THREAD

**Discussion by mikygee** [netbox-community/netbox#13457] [Help Wanted!]
Version: 3.4.8

Hello,

I'm not able to add a primary IPv4, how can I do it ?
![image](https://github.com/netbox-community/netbox/assets/2222550/281b162c-9ef5-4a94-a7f8-9776bddf0eb0)

I don't know if it's related but I cannot add interface
![image](https://github.com/netbox-community/netbox/assets/2222550/001cbe4b-0cc1-4d14-9465-d92e40a48cc7)
There is no button for that

Thank you

**Comment by candlerb** [comment]
First add an interface. Next, add an IP address to that interface. Finally, you will be able to select that address as a primary address on that device.

As a shortcut, there's a box you can check at the time of creating the IP address, which also makes it the primary at the same time.

**Comment by mikygee** [comment]
Thank you very much for your answer.
Following your explanation I managed to make it.
